routes/api/v1/dashboards.js

```diff
@@ -0,0 +1,69 @@
+const Dashboard = require('../../../models/dashboards.js')
+const { Widget } = require('../../../models/widgets.js')
+
+const express = require('express')
+const router = express.Router()
+
+router.get('/', (request, response, next) => {
+  Dashboard.find().exec()
+    .then( dashboard => response.json( dashboard ) )
+    .catch( error => next( error ) )
+})
+
+router.post('/', (request, response, next) => {
+  Dashboard.create( request.body )
+    .then( dashboard => response.json( dashboard ) )
+    .catch( error => next( error ) )
+})
+
+router.get('/:id', (request, response, next) => {
+  Dashboard.findById( request.params.id ).exec()
+    .then( dashboard => response.json( dashboard ) )
+    .catch( error => next( error ) )
+})
+
+router.put('/:id', (request, response, next) => {
+  Dashboard.findByIdAndUpdate( request.params.id, request.body ).exec()
```

Comment:
We tested this route and it seems that we can't put any old thing into the database. This validation that you're asking for sounds familiar like you've asked for this from me before but I can't find or think of where you did so. We'll probably tap you for help on this, we don't understand exactly what you're asking for.